services/QuillLMS/app/models/user.rb

```diff
@@ -233,6 +235,29 @@ def self.valid_email?(email)
     ValidatesEmailFormatOf.validate_email_format(email).nil?
   end
 
+  def require_email_verification
+    create_user_email_verification unless user_email_verification
+  end
+
+  def requires_email_verification?
+    user_email_verification.present?
+  end
+
+  def email_verified?
+    user_email_verification.present? && user_email_verification.verified?
+  end
+
+  def email_verification_pending?
+    requires_email_verification? && !email_verified?
+  end
+
+  def verify_email(verification_method, verification_token = nil)
+    # Set up email verification records if they don't exist yet
```

Comment:
what about rewriting 255-258 something like:

```
  user_email_verification
    .find_or_create_by(...)
    .verify(verification_method, verification_token)
```

And then you can get rid of the `require_email_verification` method

Reply:
That refactor isn't unreasonable, but we actually want the `require_email_verification` for other reasons.  Specifically, we want, in future PRs, the ability to create a user and flag it as needing future email verification.  Having a convenience method that you can call on `User` to establish that state will be useful for those cases.  Given that we want to keep the function, do you feel like it makes sense to avoid using it here and refactor the query, or keep things as-is?

Reply:
That makes sense.  I didn't see `require_email_verification` called outside of user but if you're going to need it later, then I would ignore my suggestion.